### PR TITLE
AddPath routes have same PathId (same localId). #2079.

### PR DESCRIPTION
**Pull request by kishiguro**
Add lock to type Bitmap.

### CODE REVIEW

File: internal/pkg/table/path.go [hunk 1]
```diff
@@ -69,19 +80,22 @@ func (b *Bitmap) FindandSetZeroBit() (uint, error) {
...
 	b.bitmap = new
+	b.m.Unlock()
 }
 
 func NewBitmap(size int) *Bitmap {
 	b := &Bitmap{}
 	if size != 0 {
 		b.bitmap = make([]uint64, (size+64-1)/64)
 	}
+	b.m = &sync.Mutex{}
```
Comment (cstockton) [inline internal/pkg/table/path.go]: See comment above.

File: internal/pkg/table/path.go [hunk 2]
```diff
@@ -36,21 +37,31 @@ const (
 
 type Bitmap struct {
 	bitmap []uint64
+	m      *sync.Mutex
```
Comment (cstockton) [inline internal/pkg/table/path.go]: This should be the zero value
```suggestion
	m      sync.Mutex
```

File: internal/pkg/table/path.go [hunk 3]
```diff
@@ -69,19 +80,22 @@ func (b *Bitmap) FindandSetZeroBit() (uint, error) {
 }
 
 func (b *Bitmap) Expand() {
+	b.m.Lock()
 	old := b.bitmap
```
Comment (cstockton) [inline internal/pkg/table/path.go]: This appears to be a basic copy operation.
```Go
cpy := make([]uint64, len(b.bitmap)+1)
copy(cpy, b.bitmap)
b.bitmap = cpy
```

**Comment by cstockton** [comment]
What is the rationale of this change?

The primary holder of Bitmap is Destination- which is also not thread safe. Most all Bitmap operations are read operations right? My concern here is this could add a lot of contention for read operations, I think it should be a RWMutex at the very least. However I have a feeling if it's to fix a race condition then others will remain dormant after.

**Comment by fujita** [comment]
I close this because this change surely fixes the reported issue. I'm still waiting for your response on the issue as I commented.

**Pull request closed without merging**